src/Bicep.LangServer/Server.cs

```diff
@@ -60,12 +62,8 @@ public async Task Run(CancellationToken cancellationToken)
             await server.WaitForExit;
         }
 
-        private static void RegisterServices(IServiceCollection services)
+        private static void RegisterServices(ResourceTypeRegistrar resourceTypeRegistrar, IServiceCollection services)
         {
-            // We don't want to recreate this every compilation.
-            // I'm not using DI for it here, because I think it makes sense to set this up for the CLI at the same time.
-            var resourceTypeRegistrar = new ResourceTypeRegistrar(new AzResourceTypeProvider());
```

Comment:
>resourceTypeRegistrar [](start = 16, length = 21)

The scope of the DI container is the language server process, so why can't you just Register AzResourceTypeProvider and ResourceTypeRegistrar in the container? When we need more providers there's a way to make it work DI as well.